appinventor/components/src/com/google/appinventor/components/runtime/FloatingActionButton.java

```diff
@@ -0,0 +1,80 @@
+package com.google.appinventor.components.runtime;
+
+import android.content.res.ColorStateList;
+import android.graphics.Color;
+import android.graphics.drawable.Drawable;
+import android.util.Log;
+import android.view.View;
+import androidx.core.widget.ImageViewCompat;
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleEvent;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+import com.google.appinventor.components.runtime.util.MediaUtil;
+
+import java.io.IOException;
+
+@DesignerComponent(version = YaVersion.FLOATING_ACTION_BUTTON_COMPONENT_VERSION,
+        category = ComponentCategory.USERINTERFACE,
+        description = " ",
+        showOnPalette = false)
+@SimpleObject
+public class FloatingActionButton extends AndroidViewComponent{
```

Comment:
Add a javadoc to this class. It will be used to generate the user-facing documentation so you should explain what the FAB is used for, how to customize it, respond to it programmatically.

Reply:
Indentation should be +2 per scope in the class body.